system-addon/content-src/components/Base/Base.jsx

```diff
@@ -1,10 +1,61 @@
 const React = require("react");
+const {connect} = require("react-redux");
+const {addLocaleData, IntlProvider} = require("react-intl");
 const TopSites = require("content-src/components/TopSites/TopSites");
 const Search = require("content-src/components/Search/Search");
 
-const Base = () => (<div className="outer-wrapper"><main>
-  <Search />
-  <TopSites />
-</main></div>);
+// Locales that should be displayed RTL
+const RTL_LIST = ["ar", "he", "fa", "ur"];
 
-module.exports = Base;
+// Add the locale data for pluralization and relative-time formatting for now,
+// this just uses english locale data. We can make this more sophisticated if
+// more features are needed.
+function addLocaleDataForReactIntl({locale}) {
+  addLocaleData([{locale, parentLocale: "en"}]);
+  document.documentElement.lang = locale;
+  document.documentElement.dir = RTL_LIST.indexOf(locale.split("-")[0]) >= 0 ? "rtl" : "ltr";
+}
+
+class Base extends React.Component {
+  componentDidMount() {
+    // Also wait for the preloaded page to show, so the tab's title updates
+    addEventListener("visibilitychange", this);
+  }
+  componentWillUpdate({App}) {
+    if (App.locale !== this.props.App.locale) {
+      addLocaleDataForReactIntl(App);
+      this.updateTitle(App);
+    }
+  }
+
+  handleEvent({type}) {
```

Comment:
Oh I think this should be okay. I was thinking of the jsx-no-bind rule, but that's more of creating callbacks from code related to `render`. Here the callback is created and used after first mount. I'll fix.